docs-src/go/chapter-durable-execution/non-deterministic-code-changes.md

```diff
@@ -60,6 +60,10 @@ Let's take a closer look:
 - `WorkflowTaskCompleted`: This Event indicates that the Worker suspended execution and made as much progress that it could.
 - `ActivityTaskScheduled`: This Event indicates that the ExecuteActivity API was called and the Worker sent the [`ScheduleActivityTask`](/references/commands#scheduleactivitytask) Command to the Server.
 - `ActivityTaskStarted`: This Event indicates that the Worker successfully polled the Activity Task and started evaluating Activity code.
+  The ActivityTaskStarted event is written into Workflow history when your activity completes or fails after all of its retries.
+  It may be counter-intuitive that this happens after the terminal Event (like [ActivityTaskCompleted](/references/events#activitytaskcompleted) or [ActivityTaskFailed](/references/events#activitytaskfailed)).
```

Comment:
What is a "terminal" Event?
I suggest trying a different descriptor because "Terminated" is an actual status.